hack/cleane2e/cleane2e.go

```diff
@@ -28,75 +46,96 @@ func main() {
 }
 
 func run(ctx context.Context, log *logrus.Entry) error {
-	subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+
+	subscriptionID = os.Getenv("AZURE_SUBSCRIPTION_ID")
+	tenantID = os.Getenv("AZURE_TENANT_ID")
+
+	ttl, err := time.ParseDuration(os.Getenv("AZURE_PURGE_TTL"))
+	if err != nil {
+		// in case of error the default ttl is always 48 hours
+		ttl = defaultTTL * time.Hour
+		log.Errorf("Cannot convert TTL to int: %s", err)
+	}
+
+	doNotTouchTag = os.Getenv("AZURE_KEEP_TAG")
+	if doNotTouchTag == "" {
+		doNotTouchTag = defaultKeepTag
+	}
 
 	authorizer, err := auth.NewAuthorizerFromEnvironment()
 	if err != nil {
-		return err
+		log.Fatal(err)
+	}
+
+	dryRun, err := strconv.ParseBool(os.Getenv("AZURE_PURGE_DRY_RUN"))
+	if err != nil {
+		log.Errorf("Cannot determine dryRun, defaulting to false: %s", err)
+	}
+	if dryRun {
+		log.Info("Dry run!")
+	}
+
+	rc := purge.ResourceCleaner{
```

Comment:
Move all these internals into `purge.New(ctx, log, subscriptionId, authorizer, dryRun)` so we don't need to manipulate with internals in entrypoint file. DeleteCheck is internals too. This makes ResourceCleaner cleaner and just internal representation.

Reply:
Damn, I forgot to write it. THanks for the pointer. Fixed :)